workspace/resource_repo.go

```diff
@@ -0,0 +1,216 @@
+package workspace
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ReposAPI exposes the Repos API
+type ReposAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// NewReposAPI creates ReposAPI instance from provider meta
+func NewReposAPI(ctx context.Context, m interface{}) ReposAPI {
+	return ReposAPI{m.(*common.DatabricksClient), ctx}
+}
+
+// ReposResponse provides information about given repository
+type ReposResponse struct {
```

Comment:
Probably it's better named as Repo or RepoSnapshot, then it fits onto reflect resource